File: Python/summaryRanges.py


nums = [0,2,3,4,6,8,9]

def summaryRanges(nums):

    if not nums: return []

    arr = []
    start = nums[0]

    for x in range(1,len(nums) + 1):

        if x == len(nums) or nums[x] != nums[x - 1] + 1:
            if start == nums[x - 1]:
                arr.append(f"{start}")
            else: 
                arr.append(f"{start} -> {nums[x - 1]}")
            if x < len(nums):    
                start = nums[x]

    return arr




print(summaryRanges(nums))    

File: JavaScript/summaryRanges.js


let nums = [0,2,3,4,6,8,9]

var summaryRanges = function(nums) {

    if (nums.length === 0) return [];
    
    let result = [];
    let start = nums[0]; 

    for (let i = 1; i <= nums.length; i++) {
       
        if (i === nums.length || nums[i] !== nums[i - 1] + 1) {
            if (start === nums[i - 1]) {
                result.push(`${start}`); 
            } else {
                result.push(`${start}->${nums[i - 1]}`); 
            }
            
            start = nums[i];
        }
    }

    return result;

};

console.log(summaryRanges(nums))